### PR TITLE
Add pkcs11-tool and mmc

### PR DESCRIPTION
**Pull request by petegriffin**
Now that meta-ts has pkcs11 TA enabled, we can use this with pkcs11-tool. Tested with the following commands.

sudo pkcs11-tool --module /usr/lib/libckteec.so.0 --init-token --label testtoken --so-pin 12341234
Using slot 0 with a present token (0x0)
Token successfully initialized

sudo pkcs11-tool --module /usr/lib/libckteec.so.0 --label test-token --so-pin 12341234 --init-pin --pin 12341234

You can also test pkcs11 ta with Xtest
sudo xtest -t pkcs11
ledge-qemuarm64:~$ sudo xtest -t pkcs11
Run test suite with level=0

TEE test application started over default TEE instance
######################################################
#
# pkcs11
#
######################################################
 
* pkcs11_1000 Initialize and close Cryptoki library
  pkcs11_1000 OK
 
* pkcs11_1001 PKCS11: List PKCS#11 slots and get information from
o pkcs11_1001.1 Test C_GetFunctionList()
  pkcs11_1001.1 OK
o pkcs11_1001.2 Test C_GetInfo()
  pkcs11_1001.2 OK
o pkcs11_1001.3 Test C_GetSlotList()
  pkcs11_1001.3 OK
o pkcs11_1001.4 Test C_Get{Slot|Token}Info()
  pkcs11_1001.4 OK
o pkcs11_1001.5 Test C_GetMechanism{List|Info}()
  pkcs11_1001.5 OK
o pkcs11_1001.6 Test C_GetMechanismList() with larger result buffer
  pkcs11_1001.6 OK
o pkcs11_1001.7 Test C_Get*Info() with invalid reference
  pkcs11_1001.7 OK
  pkcs11_1001 OK
 
* pkcs11_1002 PKCS11: Open and close PKCS#11 sessions
o pkcs11_1002.1 Test C_OpenSession()/C_GetSessionInfo()
  pkcs11_1002.1 OK
o pkcs11_1002.2 Test C_CloseSession()
  pkcs11_1002.2 OK
  pkcs11_1002 OK
 
* pkcs11_1003 PKCS11: Login to PKCS#11 token
o pkcs11_1003.1 C_InitToken() on initialized token
  pkcs11_1003.1 OK
o pkcs11_1003.2 Test C_Login()/C_Logout()
  pkcs11_1003.2 OK
o pkcs11_1003.3 Test C_SetPIN() user_type 1
  pkcs11_1003.3 OK
o pkcs11_1003.4 Test C_SetPIN() user_type 0
  pkcs11_1003.4 OK
o pkcs11_1003.5 Test C_SetPIN() user_type 2
  pkcs11_1003.5 OK
  pkcs11_1003 OK
 
* pkcs11_1004 PKCS11: create/destroy PKCS#11 simple objects
o pkcs11_1004.1 Create and destroy a volatile object
  pkcs11_1004.1 OK
o pkcs11_1004.2 Create and destroy a persistent object
  pkcs11_1004.2 OK
o pkcs11_1004.3 Create and destroy many session objects
    created object count: 61
  pkcs11_1004.3 OK
o pkcs11_1004.4 Create objects in a read-only session
  pkcs11_1004.4 OK
o pkcs11_1004.5 Create objects in a read/write session
  pkcs11_1004.5 OK
  pkcs11_1004 OK
 
* pkcs11_1005 PKCS11: Check ciphering with valid and invalid keys #1
o pkcs11_1005.1 valid usage #0
  pkcs11_1005.1 OK
o pkcs11_1005.2 valid usage #1
  pkcs11_1005.2 OK
o pkcs11_1005.3 valid usage #2
  pkcs11_1005.3 OK
o pkcs11_1005.4 valid usage #3
  pkcs11_1005.4 OK
o pkcs11_1005.5 invalid usage #0
  pkcs11_1005.5 OK
o pkcs11_1005.6 invalid usage #1
  pkcs11_1005.6 OK
o pkcs11_1005.7 invalid usage #2
  pkcs11_1005.7 OK
o pkcs11_1005.8 invalid usage #3
  pkcs11_1005.8 OK
  pkcs11_1005 OK
 
* pkcs11_1006 PKCS11: Check ciphering with valid and invalid keys #2
  pkcs11_1006 OK
 
* pkcs11_1007 PKCS11: Check operations release at session closure
E/TA:  allocate_tee_operation:262 TEE_AllocateOp. failed 0x10000010 0 0x80
    created sessions count: 31
  pkcs11_1007 OK
 
* pkcs11_1008 PKCS11: Check Compliance of C_Sign - HMAC algorithms
o pkcs11_1008.1 Sign case 0 algo (CKM_AES_CMAC)
  pkcs11_1008.1 OK
o pkcs11_1008.2 Sign case 1 algo (CKM_AES_CMAC)
  pkcs11_1008.2 OK
o pkcs11_1008.3 Sign case 2 algo (CKM_AES_CMAC)
  pkcs11_1008.3 OK
o pkcs11_1008.4 Sign case 3 algo (CKM_MD5_HMAC)
  pkcs11_1008.4 OK
o pkcs11_1008.5 Sign case 4 algo (CKM_SHA_1_HMAC)
  pkcs11_1008.5 OK
o pkcs11_1008.6 Sign case 5 algo (CKM_SHA224_HMAC)
  pkcs11_1008.6 OK
o pkcs11_1008.7 Sign case 6 algo (CKM_SHA256_HMAC)
  pkcs11_1008.7 OK
o pkcs11_1008.8 Sign case 7 algo (CKM_SHA256_HMAC)
  pkcs11_1008.8 OK
o pkcs11_1008.9 Sign case 8 algo (CKM_SHA384_HMAC)
  pkcs11_1008.9 OK
o pkcs11_1008.10 Sign case 9 algo (CKM_SHA512_HMAC)
  pkcs11_1008.10 OK
o pkcs11_1008.11 Sign case 10 algo (CKM_AES_CMAC_GENERAL)
  pkcs11_1008.11 OK
o pkcs11_1008.12 Sign case 11 algo (CKM_AES_CMAC_GENERAL)
  pkcs11_1008.12 OK
o pkcs11_1008.13 Sign case 12 algo (CKM_AES_CMAC_GENERAL)
  pkcs11_1008.13 OK
o pkcs11_1008.14 Sign case 13 algo (CKM_MD5_HMAC_GENERAL)
  pkcs11_1008.14 OK
o pkcs11_1008.15 Sign case 14 algo (CKM_SHA_1_HMAC_GENERAL)
  pkcs11_1008.15 OK
o pkcs11_1008.16 Sign case 15 algo (CKM_SHA224_HMAC_GENERAL)
  pkcs11_1008.16 OK
o pkcs11_1008.17 Sign case 16 algo (CKM_SHA256_HMAC_GENERAL)
  pkcs11_1008.17 OK
o pkcs11_1008.18 Sign case 17 algo (CKM_SHA256_HMAC_GENERAL)
  pkcs11_1008.18 OK
o pkcs11_1008.19 Sign case 18 algo (CKM_SHA384_HMAC_GENERAL)
  pkcs11_1008.19 OK
o pkcs11_1008.20 Sign case 19 algo (CKM_SHA512_HMAC_GENERAL)
  pkcs11_1008.20 OK
  pkcs11_1008 OK
 
* pkcs11_1009 PKCS11: Check Compliance of C_Verify - HMAC Algorithms
o pkcs11_1009.1 Verify case 0 algo (CKM_AES_CMAC)
  pkcs11_1009.1 OK
o pkcs11_1009.2 Verify case 1 algo (CKM_AES_CMAC)
  pkcs11_1009.2 OK
o pkcs11_1009.3 Verify case 2 algo (CKM_AES_CMAC)
  pkcs11_1009.3 OK
o pkcs11_1009.4 Verify case 3 algo (CKM_MD5_HMAC)
  pkcs11_1009.4 OK
o pkcs11_1009.5 Verify case 4 algo (CKM_SHA_1_HMAC)
  pkcs11_1009.5 OK
o pkcs11_1009.6 Verify case 5 algo (CKM_SHA224_HMAC)
  pkcs11_1009.6 OK
o pkcs11_1009.7 Verify case 6 algo (CKM_SHA256_HMAC)
  pkcs11_1009.7 OK
o pkcs11_1009.8 Verify case 7 algo (CKM_SHA256_HMAC)
  pkcs11_1009.8 OK
o pkcs11_1009.9 Verify case 8 algo (CKM_SHA384_HMAC)
  pkcs11_1009.9 OK
o pkcs11_1009.10 Verify case 9 algo (CKM_SHA512_HMAC)
  pkcs11_1009.10 OK
o pkcs11_1009.11 Verify case 10 algo (CKM_AES_CMAC_GENERAL)
  pkcs11_1009.11 OK
o pkcs11_1009.12 Verify case 11 algo (CKM_AES_CMAC_GENERAL)
  pkcs11_1009.12 OK
o pkcs11_1009.13 Verify case 12 algo (CKM_AES_CMAC_GENERAL)
  pkcs11_1009.13 OK
o pkcs11_1009.14 Verify case 13 algo (CKM_MD5_HMAC_GENERAL)
E/TA:  step_symm_operation:922 Invalid signature size: 0
E/TA:  step_symm_operation:922 Invalid signature size: 65
  pkcs11_1009.14 OK
o pkcs11_1009.15 Verify case 14 algo (CKM_SHA_1_HMAC_GENERAL)
E/TA:  step_symm_operation:922 Invalid signature size: 0
E/TA:  step_symm_operation:922 Invalid signature size: 65
  pkcs11_1009.15 OK
o pkcs11_1009.16 Verify case 15 algo (CKM_SHA224_HMAC_GENERAL)
E/TA:  step_symm_operation:922 Invalid signature size: 0
E/TA:  step_symm_operation:922 Invalid signature size: 65
  pkcs11_1009.16 OK
o pkcs11_1009.17 Verify case 16 algo (CKM_SHA256_HMAC_GENERAL)
E/TA:  step_symm_operation:922 Invalid signature size: 0
E/TA:  step_symm_operation:922 Invalid signature size: 65
  pkcs11_1009.17 OK
o pkcs11_1009.18 Verify case 17 algo (CKM_SHA256_HMAC_GENERAL)
E/TA:  step_symm_operation:922 Invalid signature size: 0
E/TA:  step_symm_operation:922 Invalid signature size: 65
  pkcs11_1009.18 OK
o pkcs11_1009.19 Verify case 18 algo (CKM_SHA384_HMAC_GENERAL)
E/TA:  step_symm_operation:922 Invalid signature size: 0
E/TA:  step_symm_operation:922 Invalid signature size: 65
  pkcs11_1009.19 OK
o pkcs11_1009.20 Verify case 19 algo (CKM_SHA512_HMAC_GENERAL)
E/TA:  step_symm_operation:922 Invalid signature size: 0
E/TA:  step_symm_operation:922 Invalid signature size: 65
  pkcs11_1009.20 OK
  pkcs11_1009 OK
 
* pkcs11_1010 PKCS11: Key Generation
o pkcs11_1010.1 Generate Secret Key with Invalid Templates
  pkcs11_1010.1 OK
o pkcs11_1010.2 Generate Generic Secret Key - Try Encrypting
  pkcs11_1010.2 OK
o pkcs11_1010.3 Generate Generic Secret Key - Try Signing
  pkcs11_1010.3 OK
o pkcs11_1010.4 Generate AES Key - Try Encrypting
  pkcs11_1010.4 OK
  pkcs11_1010 OK
 
* pkcs11_1011 PKCS11: Test Find Objects
o pkcs11_1011.1 Find created Data objects when logged in
  pkcs11_1011.1 OK
o pkcs11_1011.2 Find all objects by passing NULL template
  pkcs11_1011.2 OK
o pkcs11_1011.3 Initiate and finalize straight a search
  pkcs11_1011.3 OK
o pkcs11_1011.4 Find created Data objects when logged out
  pkcs11_1011.4 OK
o pkcs11_1011.5 Find objects 1 by 1 and match handles
  pkcs11_1011.5 OK
o pkcs11_1011.6 Find persistent objects
  pkcs11_1011.6 OK
o pkcs11_1011.7 Find session objects
  pkcs11_1011.7 OK
o pkcs11_1011.8 Login again and find Data objects
  pkcs11_1011.8 OK
o pkcs11_1011.9 Find objects from brand new session
  pkcs11_1011.9 OK
o pkcs11_1011.10 Invalid cases
E/TA:  entry_find_objects_init:531 Active object search already in progress
  pkcs11_1011.10 OK
  pkcs11_1011 OK
 
* pkcs11_1012 PKCS11: Serializer tests
o pkcs11_1012.1 Get Attribute - boolean + class
  pkcs11_1012.1 OK
o pkcs11_1012.2 Get Attribute - class + boolean
  pkcs11_1012.2 OK
o pkcs11_1012.3 Get Attribute - query size boolean + class
  pkcs11_1012.3 OK
o pkcs11_1012.4 Get Attribute - query size class + boolean
  pkcs11_1012.4 OK
o pkcs11_1012.5 Get Attribute - value with exact size buffer
  pkcs11_1012.5 OK
o pkcs11_1012.6 Get Attribute - value with larger buffer
  pkcs11_1012.6 OK
o pkcs11_1012.7 Get Attribute - bool + value with exact size + class + bool
  pkcs11_1012.7 OK
o pkcs11_1012.8 Get Attribute - bool + value with larger buffer + class + bool
  pkcs11_1012.8 OK
o pkcs11_1012.9 Get Attribute - allowed mechanism list
  pkcs11_1012.9 OK
  pkcs11_1012 OK
 
* pkcs11_1013 PKCS11: Object creation upon session type
o pkcs11_1013.1 Create objects in R/O Public Session
  pkcs11_1013.1 OK
o pkcs11_1013.2 Create objects in R/O Public Session
  pkcs11_1013.2 OK
o pkcs11_1013.3 Create objects in R/O User Session
  pkcs11_1013.3 OK
o pkcs11_1013.4 Create objects in R/W User Session
  pkcs11_1013.4 OK
o pkcs11_1013.5 Create objects in R/W SO Session
  pkcs11_1013.5 OK
  pkcs11_1013 OK
 
* pkcs11_1014 PKCS11: Test C_SetAttributeValue()
o pkcs11_1014.1 Set attributes on secret key object
  pkcs11_1014.1 OK
o pkcs11_1014.2 Test Invalid template with R/O Attributes
  pkcs11_1014.2 OK
o pkcs11_1014.3 Test Invalid template with Invalid Attribute
  pkcs11_1014.3 OK
o pkcs11_1014.4 Set CKA_TRUSTED with and w/o SO Login
  pkcs11_1014.4 OK
  pkcs11_1014 OK
 
* pkcs11_1015 PKCS11: Test C_CopyObject()
o pkcs11_1015.1 Copy Local Obj with NULL Template
  pkcs11_1015.1 OK
o pkcs11_1015.2 Copy a non-local object with NULL Template
  pkcs11_1015.2 OK
o pkcs11_1015.3 Copy Object with Valid Template
  pkcs11_1015.3 OK
o pkcs11_1015.4 Copy Object as a Private Object
  pkcs11_1015.4 OK
o pkcs11_1015.5 Copy Object with Invalid Template
  pkcs11_1015.5 OK
o pkcs11_1015.6 Copy Object with COPYABLE false
  pkcs11_1015.6 OK
o pkcs11_1015.7 Copy session object to token in RO session
  pkcs11_1015.7 OK
  pkcs11_1015 OK
 
* pkcs11_1016 PKCS11: Random number generator tests
o pkcs11_1016.1 Seed random bytes
  pkcs11_1016.1 OK
o pkcs11_1016.2 Seed random bytes with zero length buffer
  pkcs11_1016.2 OK
o pkcs11_1016.3 Generate random bytes
  pkcs11_1016.3 OK
o pkcs11_1016.4 Generate random bytes with zero length buffer
  pkcs11_1016.4 OK
  pkcs11_1016 OK
 
* pkcs11_1017 PKCS11: AES Key Derivation tests
o pkcs11_1017.1 Derive Generic secret - AES-ECB Mechanism
  pkcs11_1017.1 OK
o pkcs11_1017.2 Derive AES key using AES-EBC
  pkcs11_1017.2 OK
o pkcs11_1017.3 Derive AES key using AES-CBC
  pkcs11_1017.3 OK
o pkcs11_1017.4 Derive Generic secret key using AES-CBC
  pkcs11_1017.4 OK
o pkcs11_1017.5 Invalid parameters during derivation
E/TA:  check_mechanism_against_processing:140 Processing 0x1081/AES_ECB not permitted (3/0)
  pkcs11_1017.5 OK
o pkcs11_1017.6 Failure if operation already active
  pkcs11_1017.6 OK
o pkcs11_1017.7 Failure if parent key CKA_ENCRYPT is TRUE
  pkcs11_1017.7 OK
o pkcs11_1017.8 Failure if parent key CKA_DERIVE is FALSE
  pkcs11_1017.8 OK
  pkcs11_1017 OK
 
* pkcs11_1018 PKCS11: Digest tests
o pkcs11_1018.1 Simple digest tests - update - CKM_MD5/empty
  pkcs11_1018.1 OK
o pkcs11_1018.2 Simple digest tests - oneshot - CKM_MD5/empty
  pkcs11_1018.2 OK
o pkcs11_1018.3 Simple digest tests - AES key update - SHA-256
  pkcs11_1018.3 OK
o pkcs11_1018.4 Simple digest tests - generic secret key update - SHA-256
  pkcs11_1018.4 OK
o pkcs11_1018.5 Query digest size - C_DigestFinal
  pkcs11_1018.5 OK
o pkcs11_1018.6 Query digest size - C_Digest
  pkcs11_1018.6 OK
o pkcs11_1018.7 Query digest size - buffer too small
  pkcs11_1018.7 OK
o pkcs11_1018.8 Test bad arguments - C_DigestUpdate
  pkcs11_1018.8 OK
o pkcs11_1018.9 Test bad arguments - C_DigestFinal with NULL digest
  pkcs11_1018.9 OK
o pkcs11_1018.10 Test bad arguments - C_DigestFinal with digest but NULL size
  pkcs11_1018.10 OK
o pkcs11_1018.11 Test bad arguments - C_Digest with NULL data but non-zero size
  pkcs11_1018.11 OK
o pkcs11_1018.12 Test bad arguments - C_Digest with NULL digest
  pkcs11_1018.12 OK
o pkcs11_1018.13 Test bad arguments - C_DigestFinal with digest but NULL size
  pkcs11_1018.13 OK
o pkcs11_1018.14 Test bad arguments - C_DigestKey with invalid key handle
  pkcs11_1018.14 OK
o pkcs11_1018.15 Test bad arguments - C_DigestKey with non-secret key type
  pkcs11_1018.15 OK
  pkcs11_1018 OK
 
* pkcs11_1019 PKCS11: Elliptic Curve key generation and signing
o pkcs11_1019.1 P-256: Generate key pair
  pkcs11_1019.1 OK
o pkcs11_1019.2 P-256: Sign & verify tests - oneshot - CKM_ECDSA
  pkcs11_1019.2 OK
o pkcs11_1019.3 P-256: Sign & verify - oneshot - CKM_ECDSA_SHA1
  pkcs11_1019.3 OK
o pkcs11_1019.4 P-256: Sign & verify - oneshot - CKM_ECDSA_SHA224
  pkcs11_1019.4 OK
o pkcs11_1019.5 P-256: Sign & verify - oneshot - CKM_ECDSA_SHA256
  pkcs11_1019.5 OK
o pkcs11_1019.6 P-256: Sign & verify - oneshot - CKM_ECDSA_SHA384
  pkcs11_1019.6 OK
o pkcs11_1019.7 P-256: Sign & verify - oneshot - CKM_ECDSA_SHA512
  pkcs11_1019.7 OK
o pkcs11_1019.8 P-256: Destroy keys
  pkcs11_1019.8 OK
o pkcs11_1019.9 P-384: Generate key pair
  pkcs11_1019.9 OK
o pkcs11_1019.10 P-384: Sign & verify tests - oneshot - CKM_ECDSA
  pkcs11_1019.10 OK
o pkcs11_1019.11 P-384: Sign & verify - oneshot - CKM_ECDSA_SHA1
  pkcs11_1019.11 OK
o pkcs11_1019.12 P-384: Sign & verify - oneshot - CKM_ECDSA_SHA224
  pkcs11_1019.12 OK
o pkcs11_1019.13 P-384: Sign & verify - oneshot - CKM_ECDSA_SHA256
  pkcs11_1019.13 OK
o pkcs11_1019.14 P-384: Sign & verify - oneshot - CKM_ECDSA_SHA384
  pkcs11_1019.14 OK
o pkcs11_1019.15 P-384: Sign & verify - oneshot - CKM_ECDSA_SHA512
  pkcs11_1019.15 OK
o pkcs11_1019.16 P-384: Destroy keys
  pkcs11_1019.16 OK
  pkcs11_1019 OK
 
* pkcs11_1020 PKCS11: AES Key Wrap/UnWrap tests
o pkcs11_1020.1 Test key wrap with AES ECB
  pkcs11_1020.1 OK
o pkcs11_1020.2 Test key unwrap with AES ECB
  pkcs11_1020.2 OK
o pkcs11_1020.3 Invalid UnWrap cases
E/TA:  create_attributes_from_template:1071 Inconsistent class/type
  pkcs11_1020.3 OK
o pkcs11_1020.4 Invalid Wrap cases
E/TA:  check_mechanism_against_processing:140 Processing 0x211/MD5_HMAC not permitted (4/0)
  pkcs11_1020.4 OK
o pkcs11_1020.5 Wrap with different length key
  pkcs11_1020.5 OK
o pkcs11_1020.6 Test Wrap/Unwrap with indirect template
  pkcs11_1020.6 OK
o pkcs11_1020.7 Test usage of CKA_WRAP_WITH_TRUSTED
  pkcs11_1020.7 OK
  pkcs11_1020 OK
 
* pkcs11_1021 PKCS11: RSA PKCS key generation and signing
o pkcs11_1021.1 RSA-1024: Generate key pair
  pkcs11_1021.1 OK
o pkcs11_1021.2 RSA-1024: Sign & verify tests - oneshot - CKM_RSA_PKCS
  pkcs11_1021.2 OK
o pkcs11_1021.3 RSA-1024: Sign & verify - oneshot - CKM_MD5_RSA_PKCS
  pkcs11_1021.3 OK
o pkcs11_1021.4 RSA-1024: Sign & verify - oneshot - CKM_SHA1_RSA_PKCS
  pkcs11_1021.4 OK
o pkcs11_1021.5 RSA-1024: Sign & verify - oneshot - CKM_SHA224_RSA_PKCS
  pkcs11_1021.5 OK
o pkcs11_1021.6 RSA-1024: Sign & verify - oneshot - CKM_SHA256_RSA_PKCS
  pkcs11_1021.6 OK
o pkcs11_1021.7 RSA-1024: Sign & verify - oneshot - CKM_SHA384_RSA_PKCS
  pkcs11_1021.7 OK
o pkcs11_1021.8 RSA-1024: Sign & verify - oneshot - CKM_SHA512_RSA_PKCS
  pkcs11_1021.8 OK
o pkcs11_1021.9 RSA-2048: Generate key pair
  pkcs11_1021.9 OK
o pkcs11_1021.10 RSA-2048: Sign & verify tests - oneshot - CKM_RSA_PKCS
  pkcs11_1021.10 OK
o pkcs11_1021.11 RSA-2048: Sign & verify - oneshot - CKM_MD5_RSA_PKCS
  pkcs11_1021.11 OK
o pkcs11_1021.12 RSA-2048: Sign & verify - oneshot - CKM_SHA1_RSA_PKCS
  pkcs11_1021.12 OK
o pkcs11_1021.13 RSA-2048: Sign & verify - oneshot - CKM_SHA224_RSA_PKCS
  pkcs11_1021.13 OK
o pkcs11_1021.14 RSA-2048: Sign & verify - oneshot - CKM_SHA256_RSA_PKCS
  pkcs11_1021.14 OK
o pkcs11_1021.15 RSA-2048: Sign & verify - oneshot - CKM_SHA384_RSA_PKCS
  pkcs11_1021.15 OK
o pkcs11_1021.16 RSA-2048: Sign & verify - oneshot - CKM_SHA512_RSA_PKCS
  pkcs11_1021.16 OK
  pkcs11_1021 OK
 
* pkcs11_1022 PKCS11: RSA PSS key generation and signing
o pkcs11_1022.1 RSA-1024: Generate key pair
  pkcs11_1022.1 OK
o pkcs11_1022.2 RSA-1024: Sign & verify - oneshot - RSA-PSS/SHA1
  pkcs11_1022.2 OK
o pkcs11_1022.3 RSA-1024: Sign & verify - oneshot - RSA-PSS/SHA1/mech
  pkcs11_1022.3 OK
o pkcs11_1022.4 RSA-1024: Sign & verify - oneshot - RSA-PSS/SHA224
  pkcs11_1022.4 OK
o pkcs11_1022.5 RSA-1024: Sign & verify - oneshot - RSA-PSS/SHA224/mech
  pkcs11_1022.5 OK
o pkcs11_1022.6 RSA-1024: Sign & verify - oneshot - RSA-PSS/SHA256
  pkcs11_1022.6 OK
o pkcs11_1022.7 RSA-1024: Sign & verify - oneshot - RSA-PSS/SHA256/mech
  pkcs11_1022.7 OK
o pkcs11_1022.8 RSA-1024: Sign & verify - oneshot - RSA-PSS/SHA384
  pkcs11_1022.8 OK
o pkcs11_1022.9 RSA-1024: Sign & verify - oneshot - RSA-PSS/SHA384/mech
  pkcs11_1022.9 OK
o pkcs11_1022.10 RSA-1024: Sign & verify - oneshot - RSA-PSS/SHA512
  pkcs11_1022.10 OK
o pkcs11_1022.11 RSA-1024: Sign & verify - oneshot - RSA-PSS/SHA512/mech
  pkcs11_1022.11 OK
o pkcs11_1022.12 RSA-2048: Generate key pair
  pkcs11_1022.12 OK
o pkcs11_1022.13 RSA-2048: Sign & verify - oneshot - RSA-PSS/SHA1
  pkcs11_1022.13 OK
o pkcs11_1022.14 RSA-2048: Sign & verify - oneshot - RSA-PSS/SHA1/mech
  pkcs11_1022.14 OK
o pkcs11_1022.15 RSA-2048: Sign & verify - oneshot - RSA-PSS/SHA224
  pkcs11_1022.15 OK
o pkcs11_1022.16 RSA-2048: Sign & verify - oneshot - RSA-PSS/SHA224/mech
  pkcs11_1022.16 OK
o pkcs11_1022.17 RSA-2048: Sign & verify - oneshot - RSA-PSS/SHA256
  pkcs11_1022.17 OK
o pkcs11_1022.18 RSA-2048: Sign & verify - oneshot - RSA-PSS/SHA256/mech
  pkcs11_1022.18 OK
o pkcs11_1022.19 RSA-2048: Sign & verify - oneshot - RSA-PSS/SHA384
  pkcs11_1022.19 OK
o pkcs11_1022.20 RSA-2048: Sign & verify - oneshot - RSA-PSS/SHA384/mech
  pkcs11_1022.20 OK
o pkcs11_1022.21 RSA-2048: Sign & verify - oneshot - RSA-PSS/SHA512
  pkcs11_1022.21 OK
o pkcs11_1022.22 RSA-2048: Sign & verify - oneshot - RSA-PSS/SHA512/mech
  pkcs11_1022.22 OK
  pkcs11_1022 OK
 
* pkcs11_1023 PKCS11: RSA OAEP key generation and crypto operations
o pkcs11_1023.1 RSA-1024: Generate key pair
  pkcs11_1023.1 OK
o pkcs11_1023.2 RSA-1024: Encrypt & decrypt - oneshot - RSA-OAEP/SHA1
  pkcs11_1023.2 OK
o pkcs11_1023.3 RSA-1024: Encrypt & decrypt - oneshot - RSA-OAEP/SHA1/label
  pkcs11_1023.3 OK
o pkcs11_1023.4 RSA-1024: Encrypt & decrypt - oneshot - RSA-OAEP/SHA224
  pkcs11_1023.4 OK
o pkcs11_1023.5 RSA-1024: Encrypt & decrypt - oneshot - RSA-OAEP/SHA224/label
  pkcs11_1023.5 OK
o pkcs11_1023.6 RSA-1024: Encrypt & decrypt - oneshot - RSA-OAEP/SHA256
  pkcs11_1023.6 OK
o pkcs11_1023.7 RSA-1024: Encrypt & decrypt - oneshot - RSA-OAEP/SHA256/label
  pkcs11_1023.7 OK
o pkcs11_1023.8 RSA-1024: Encrypt & decrypt - oneshot - RSA-OAEP/SHA384
  pkcs11_1023.8 OK
o pkcs11_1023.9 RSA-1024: Encrypt & decrypt - oneshot - RSA-OAEP/SHA384/label
  pkcs11_1023.9 OK
o pkcs11_1023.10 RSA-1024: Encrypt & decrypt - oneshot - RSA-OAEP/SHA512
E/TC:? 0 crypto_acipher_rsaes_encrypt:416 rsa_encrypt_key_ex() returned 22
  pkcs11_1023.10 OK
o pkcs11_1023.11 RSA-1024: Encrypt & decrypE/TC:? 0t - oneshot - RSA-OAEP/SHA512/label
 crypto_acipher_rsaes_encrypt:416 rsa_encrypt_key_ex() returned 22
  pkcs11_1023.11 OK
o pkcs11_1023.12 RSA-2048: Generate key pair
  pkcs11_1023.12 OK
o pkcs11_1023.13 RSA-2048: Encrypt & decrypt - oneshot - RSA-OAEP/SHA1
  pkcs11_1023.13 OK
o pkcs11_1023.14 RSA-2048: Encrypt & decrypt - oneshot - RSA-OAEP/SHA1/label
  pkcs11_1023.14 OK
o pkcs11_1023.15 RSA-2048: Encrypt & decrypt - oneshot - RSA-OAEP/SHA224
  pkcs11_1023.15 OK
o pkcs11_1023.16 RSA-2048: Encrypt & decrypt - oneshot - RSA-OAEP/SHA224/label
  pkcs11_1023.16 OK
o pkcs11_1023.17 RSA-2048: Encrypt & decrypt - oneshot - RSA-OAEP/SHA256
  pkcs11_1023.17 OK
o pkcs11_1023.18 RSA-2048: Encrypt & decrypt - oneshot - RSA-OAEP/SHA256/label
  pkcs11_1023.18 OK
o pkcs11_1023.19 RSA-2048: Encrypt & decrypt - oneshot - RSA-OAEP/SHA384
  pkcs11_1023.19 OK
o pkcs11_1023.20 RSA-2048: Encrypt & decrypt - oneshot - RSA-OAEP/SHA384/label
  pkcs11_1023.20 OK
o pkcs11_1023.21 RSA-2048: Encrypt & decrypt - oneshot - RSA-OAEP/SHA512
  pkcs11_1023.21 OK
o pkcs11_1023.22 RSA-2048: Encrypt & decrypt - oneshot - RSA-OAEP/SHA512/label
  pkcs11_1023.22 OK
  pkcs11_1023 OK
 
* pkcs11_1024 PKCS11: X509 Certificate operations
o pkcs11_1024.1 Import X.509 Certificate
  pkcs11_1024.1 OK
o pkcs11_1024.2 Import X.509 Certificate with default values
  pkcs11_1024.2 OK
o pkcs11_1024.3 Try import with invalid category
E/TA:  create_certificate_attributes:674 Invalid certificate category 0xffffffff
  pkcs11_1024.3 OK
o pkcs11_1024.4 Try import with invalid category size
  pkcs11_1024.4 OK
o pkcs11_1024.5 Try import with invalid name hash alg size
  pkcs11_1024.5 OK
  pkcs11_1024 OK
+-----------------------------------------------------
Result of testsuite pkcs11:
pkcs11_1000 OK
pkcs11_1001 OK
pkcs11_1002 OK
pkcs11_1003 OK
pkcs11_1004 OK
pkcs11_1005 OK
pkcs11_1006 OK
pkcs11_1007 OK
pkcs11_1008 OK
pkcs11_1009 OK
pkcs11_1010 OK
pkcs11_1011 OK
pkcs11_1012 OK
pkcs11_1013 OK
pkcs11_1014 OK
pkcs11_1015 OK
pkcs11_1016 OK
pkcs11_1017 OK
pkcs11_1018 OK
pkcs11_1019 OK
pkcs11_1020 OK
pkcs11_1021 OK
pkcs11_1022 OK
pkcs11_1023 OK
pkcs11_1024 OK
+-----------------------------------------------------
3725 subtests of which 0 failed
25 test cases of which 0 failed
0 test cases were skipped
TEE test application done!

### CODE REVIEW

**Comment by petegriffin** [comment]
@jforissier fyi

File: meta-ledge-sw/recipes-samples/packagegroups/packagegroup-ledge-iot.bb [hunk 1]
```diff
@@ -175,6 +175,8 @@ RDEPENDS_packagegroup-ledge-iot = "\
 	xz \
 	zlib \
 	zram \
+	opensc \
+        mmc-utils \
 	"
```
Comment (apalos) [inline meta-ledge-sw/recipes-samples/packagegroups/packagegroup-ledge-iot.bb:180]: Is the indentation off or is this just a github issue?

Comment (petegriffin) [reply meta-ledge-sw/recipes-samples/packagegroups/packagegroup-ledge-iot.bb:180]: no spaces not a tab..whoops, just fixed it

**Review by apalos — APPROVED** [review]

**Comment by jforissier** [comment]
LGTM

**Comment by muvarov** [comment]
Patch is trivial but anyway waiting for CI...

File: meta-ledge-sw/recipes-samples/packagegroups/packagegroup-ledge-iot.bb (hunk 1 above)
Comment (etienne-lms) [reply meta-ledge-sw/recipes-samples/packagegroups/packagegroup-ledge-iot.bb:180]: The list seems to follow an alphabetical order. should move above.

**Review by etienne-lms — COMMENTED** [review]
nitpicking comment on recipes layout.
Otherwise LGTM.

File: meta-ledge-sw/recipes-samples/packagegroups/packagegroup-ledge-iot.bb (hunk 1 above)
Comment (petegriffin) [reply meta-ledge-sw/recipes-samples/packagegroups/packagegroup-ledge-iot.bb]: now in alphabetical order

**Comment by etienne-lms** [comment]
sorry @petegriffin, maybe commit message should also tell why **mmc-utils** is embedded (the details were in the 2nd commit in previous series).

**Comment by petegriffin** [comment]
> sorry @petegriffin, maybe commit message should also tell why **mmc-utils** is embedded (the details were in the 2nd commit in previous series).

ah whoops I made a mistake in the rebase..just fixed

**Review by etienne-lms — APPROVED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>`

**Pull request merged**